docs/deepops/configuration.md

```diff
@@ -0,0 +1,219 @@
+These variables go in files named for each host in the `host_vars` directory:
+
+```
+host_vars/
+└── example01
+```
+
+The variables in `host_vars/example01` would only be applied for a host named `example01`.
+
+
+## Adding custom playbooks
+
+In the process of setting up your cluster, you may want to write your own Ansible playbooks for features that DeepOps doesn't include!
+
+For example, DeepOps doesn't (yet) include a playbook for installing the classic Linux game [Pingus](https://pingus.seul.org/).
+If you wrote an Ansible playbook for installing Pingus on your cluster (to give your users a way to [entertain themselves while their code compiles](https://xkcd.com/303/)),
+you would put it in your `config/playbooks` directory:
+
+```
+playbooks/
+└── pingus.yml
+```
+
+This playbook can then be run with Ansible like any other DeepOps playbook:
+
+```
+ansible-playbook config/playbooks/pingus.yml
+```
+
+And can make use of variables in your `config` directory like other playboks.
+
+
+## Managing your configuration directory in Git
+
+We recommend creating a separate Git repository for managing your cluster configuration, so that you can track changes to your cluster independently of DeepOps changes and upgrades.
+(The DeepOps [`.gitignore` file](../../.gitignore) is set up to faciliate this by ignoring directories that start with the string `config`, except for `config.example`.)
+
+A good practice is to start a new Git repository by copying the example configuration:
+
+```
+$ cp -R config.example/ config.my-cluster/
+$ cd config.my-cluster/
+$ git init
+$ git commit -am "Start from the example configuration"
+```
+
+And then pushing this repository to a remote Git host, such as Github:
+
+```
+$ cd config.my-cluster/
+$ git remote add origin <your-git-remote>
+$ git push -u origin main
+```
+
+Once you have a repository set up, you can use it to track changes as you configure your cluster.
+For example, if you change the version of Slurm being installed:
+
+```
+$ cd config.my-cluster/group_vars/
+
+# Edit your Slurm configuration to upgrade to version 20.11
+$ vim slurm-cluster.yml
+
+$ git add slurm-cluster.yml
+$ git commit -m "Update cluster to Slurm 20.11"
+```
+
+
+## Using multiple configuration directories for separate clusters
+
+If you're running multiple clusters, you can keep their configuration in separate configuration directories within the same DeepOps repository clone:
+
+```
+config.cluster-0
+└── group_vars
+config.cluster-1
+├── group_vars
+└── host_vars
+config.cluster-2
+├── group_vars
+├── host_vars
+└── playbooks
+```
+
+These configuration directories can each have their own inventory files, Ansible variables, and playbooks.
+
+You can then run Ansible for each of the clusters independently by specifying the inventory file on the command line:
+
+```
+ansible-playbook -i config.cluster-1/inventory playbooks/slurm-cluster.yml
+```
+
```

Comment:
We might want to add a section on environment variables. Most of our deployment scripts have reasonable defaults, but will allow us to enable/disable features or install specific versions based of GLOBAL_ENVIRONMENT_VARIABLES.

Perhaps we suggest that they maintain a config/environment.sh file and source it before running scripts? In fact, maybe this is something we should consider adding to all of our scripts.

Reply:
I'm not sure this fits in with the rest of the doc, as a lot of it will be script-specific. However, I think documenting that better is definitely a good idea... need to think about the right way to do that.